app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainViewModel.kt

```diff
@@ -1,26 +1,29 @@
 package com.mstudio.superheroarch.presentation.overview
 
 import com.mstudio.superheroarch.presentation.model.CharacterData
+import com.mstudio.superheroarch.remotedatasource.api.RickAndMortyApiHelper
 import com.mstudio.superheroarch.remotedatasource.model.toCharacterData
 import com.mstudio.superheroarch.repository.RickAndMortyRepository
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 class MainViewModel(
-    private val view: MainViewTranslator
+    private val view: MainViewTranslator,
+    private val repository: RickAndMortyRepository = RickAndMortyRepository(api = RickAndMortyApiHelper.create()),
```

Comment:
do you need to add the repository to the constructor? you are using mockito to mock the dependencies.

Reply:
There is an inconsistency between this ViewModel and Details. In this you use a repository and in the other a useCase. I would propose both of them to be aligned.